rotkehlchen/tests/db/test_db_upgrades.py

```diff
@@ -2602,6 +2603,45 @@ def test_upgrade_db_31_to_32(user_data_dir):  # pylint: disable=unused-argument
     assert cursor.fetchone() == (1, expected_timestamp // 10)
 
 
+@pytest.mark.parametrize('use_clean_caching_directory', [True])
+def test_upgrade_db_32_to_33(user_data_dir):  # pylint: disable=unused-argument  # noqa: E501
+    """Test upgrading the DB from version 32 to version 33.
+    """
+    msg_aggregator = MessagesAggregator()
+    _use_prepared_db(user_data_dir, 'v32_rotkehlchen.db')
+    db_v32 = _init_db_with_target_version(
+        target_version=32,
+        user_data_dir=user_data_dir,
+        msg_aggregator=msg_aggregator,
+    )
+    cursor = db_v32.conn.cursor()
+    # check that you cannot add blockchain column in xpub_mappings
+    with pytest.raises(sqlcipher.OperationalError) as exc_info:  # pylint: disable=no-member
+        cursor.execute(
+            'INSERT INTO xpub_mappings(address, xpub, derivation_path, account_index, derived_index, blockchain) '  # noqa: 501
+            'VALUES ("1234", "abcd", "d", 3, 6, "BCH");',
+        )
+    assert 'cannot INSERT into generated column "blockchain"' in str(exc_info)
+    xpub_mappings = cursor.execute('SELECT * FROM xpub_mappings').fetchall()
```

Comment:
nitpick:

Can you also check that the xpub mapping before has the values you inserted in the DB and that they make sense? Make sure the values are real/sensible.